### ISSUE TITLE
ASR models with transformer as encoder and pure attention loss doesn't converge on Librispeech

### ISSUE THREAD

**Issue by ZhihaoDU** [espnet/espnet#5298]
**Describe the bug**
ASR models with transformer as encoder and pure attention loss (by set ctc_weight to 0.0) doesn't convergence on Librispeech. When I use the default config at egs2/librispeech/conf/train_asr_transformer.yaml, the mode converged, I got a log like:
```
2023-07-13 10:45:35,775 (trainer:830) INFO: 24epoch:train:1251-1300batch: iter_time=1.45692e-04, forward_time=0.44546, loss_ctc=25.42261, loss_att=12.66447, acc=0.90140, loss=16.49191, backward_time=0.67838, optim_step_time=0.02063, optim0_lr0=0.00174, train_time=1.20348
```
However, when I set ctc_weight to 0.0, aka, pure attention loss, the model doesn't converge anymore. I got a log like:
```
2023-07-13 10:40:13,592 (trainer:830) INFO: 25epoch:train:651-700batch: iter_time=1.49469e-04, forward_time=0.41950, loss_att=141.77131, acc=0.30478, loss=141.77131, backward_time=0.66452, optim_step_time=0.02028, optim0_lr0=0.00172, train_time=1.14912
```
It's really strange.

**Basic environments:**
 - OS information: Linux version 3.10.0-327.ali2012.alios7.x86_64 (gcc version 4.8.5 20150623 (Red Hat 4.8.5-4) (GCC) )
 - python version: 3.8
 - espnet version: 0.10.7a1
 - pytorch version 1.12

**Task information:**
 - Task: ASR
 - Recipe: librispeech
 - ESPnet2
 - Config: egs2/librispeech/conf/train_asr_transformer.yaml

**To Reproduce**
1. move to a recipe directory, `cd egs2/librispeech/asr1`
2. set ctc_weight in egs2/librispeech/conf/train_asr_transformer.yaml to 0.0
3. run the run.sh

**Comment by sw005320** [comment]
Please tune the learning rate.

**Comment by ZhihaoDU** [comment]
@sw005320 Thanks for your suggestion. The default learning rate is 2e-3. I have tried 2e-2, 2e-4 and 2e-5. The training process is still strange, and the model doesn't converge as well. 

with CTC loss
lr=2e-3
```
2023-07-13 03:05:09,018 (trainer:830) INFO: 8epoch:train:901-950batch: iter_time=1.41004e-04, forward_time=0.44076, loss_ctc=46.40360, loss_att=29.38356, acc=0.79233, loss=34.48957, backward_time=0.66919, optim_step_time=0.02074, optim0_lr0=8.43560e-04, train_time=1.19228
```

without CTC loss
For lr=2e-2
```
2023-07-13 16:35:08,265 (trainer:830) INFO: 8epoch:train:901-950batch: iter_time=1.69026e-04, forward_time=0.43048, loss_att=133.45856, acc=0.24677, loss=133.45856, backward_time=0.64906, optim_step_time=0.02267, optim0_lr0=0.00844, train_time=1.15213
```
For lr=2e-3
```
2023-07-13 03:01:58,687 (trainer:830) INFO: 8epoch:train:901-950batch: iter_time=1.08484e-04, forward_time=0.42519, loss_att=137.36981, acc=0.24666, loss=137.36981, backward_time=0.64676, optim_step_time=0.02064, optim0_lr0=8.43560e-04, train_time=1.14413
```
For lr=2e-4
```
2023-07-13 15:47:49,861 (trainer:830) INFO: 8epoch:train:901-950batch: iter_time=1.34621e-04, forward_time=0.42866, loss_att=155.76838, acc=0.17002, loss=155.76838, backward_time=0.64722, optim_step_time=0.02088, optim0_lr0=8.43560e-05, train_time=1.14406
```
For lr=2e-5
```
2023-07-13 15:48:51,252 (trainer:830) INFO: 8epoch:train:901-950batch: iter_time=1.38169e-04, forward_time=0.43222, loss_att=176.38136, acc=0.11624, loss=176.38136, backward_time=0.65016, optim_step_time=0.02193, optim0_lr0=8.43560e-06, train_time=1.14995
```

**Comment by sw005320** [comment]
Can you paste the learning curves?

**Comment by ZhihaoDU** [comment]
LR=2e-3, learning curves on training set
Pure Attention loss (CTC_weight=0.0)
![image](https://github.com/espnet/espnet/assets/28995683/e01faae0-451f-4b47-9225-945cc7ac604e)
![image](https://github.com/espnet/espnet/assets/28995683/399dc861-7463-4e61-ac74-ac659d993bbb)
![image](https://github.com/espnet/espnet/assets/28995683/437be15e-8a29-41bb-a4ce-2288620c2f0f)
CTC-Attention loss (CTC_weight=0.3)
![image](https://github.com/espnet/espnet/assets/28995683/2ab40c4a-d752-4dbf-bee9-7abe2ccffb7d)
![image](https://github.com/espnet/espnet/assets/28995683/52e9f84c-78d4-4f6b-81ad-b305b5333a18)
![image](https://github.com/espnet/espnet/assets/28995683/ce94badf-904d-48b1-b02a-4a7a72166ab2)

**Comment by sw005320** [comment]
Can you reduce the model size?
This makes it easy to tune.

**Comment by ZhihaoDU** [comment]
Thanks, I will try to reduce the attention_dim from 512 to 256.
